### ISSUE TITLE
Better mouse support on OS X

### ISSUE THREAD

**Issue by rennis250** [mawww/kakoune#608]
The mouse doesn't work as smoothly on OS X as it does on Linux. I can re-position the cursor by clicking, but I cannot do the following:

- select regions of text
- scroll the window

This applies for kak in Terminal.app as well as iTerm.

Best,
Rob

**Comment by rennis250** [comment]
What I should say is that on OS X, you can only scroll upwards, not downwards.

**Comment by lenormf** [comment]
Those sound like issues with ncurses, which version do you use ?

**Comment by rennis250** [comment]
OS X 10.11.3 comes with libncurses version 5.4, so that will be the newest version that any OS X user could have by default. I will try installing version 6.0 and will let you know.

**Comment by rennis250** [comment]
Yes, that was the reason. It works now. We should mention this in the README and update the Homebrew formula. Basically, the user needs to add the "homebrew/dupes" tap and then install "homebrew/dupes/ncurses", which gives version 6.0 of ncurses from the GNU page. After that, the following compiler flags need to be added to the Kakoune build process, after line 39 of the Makefile (the Darwin section):

```
CPPFLAGS += -I/usr/local/opt/ncurses/include
LDFLAGS += -L/usr/local/opt/ncurses/lib
```

**Comment by lenormf** [comment]
Pull requests are welcome!
